extras/indexing/src/main/java/org/apache/rya/api/client/accumulo/AccumuloDeletePeriodicPCJ.java

```diff
@@ -0,0 +1,131 @@
+        final boolean ryaInstanceExists = originalDetails.isPresent();
+        if(!ryaInstanceExists) {
+            throw new InstanceDoesNotExistException(String.format("The '%s' instance of Rya does not exist.", instanceName));
+        }
+
+        final boolean pcjIndexingEnabled = originalDetails.get().getPCJIndexDetails().isEnabled();
+        if(!pcjIndexingEnabled) {
+            throw new RyaClientException(String.format("The '%s' instance of Rya does not have PCJ Indexing enabled.", instanceName));
+        }
+
+        // If the PCJ was being maintained by a Fluo application, then stop that process.
+        final PCJIndexDetails pcjIndexDetails = originalDetails.get().getPCJIndexDetails();
+        final Optional<FluoDetails> fluoDetailsHolder = pcjIndexDetails.getFluoDetails();
+
+        if (fluoDetailsHolder.isPresent()) {
+            final String fluoAppName = pcjIndexDetails.getFluoDetails().get().getUpdateAppName();
+            try {
+                stopUpdatingPCJ(instanceName, fluoAppName, pcjId, topic, brokers);
+            } catch (MalformedQueryException | UnsupportedQueryException | QueryDeletionException e) {
+                throw new RyaClientException(String.format("Unable to delete Periodic Query with id: %s", pcjId), e);
+            }
+        } else {
+            log.error(String.format("Could not stop the Fluo application from updating the PCJ because the Fluo Details are "
+                    + "missing for the Rya instance named '%s'.", instanceName));
+        }
+        
+    }
+
+
+    private void stopUpdatingPCJ(final String ryaInstance, final String fluoAppName, final String pcjId, final String topic, final String brokers) throws UnsupportedQueryException, MalformedQueryException, QueryDeletionException {
+        requireNonNull(fluoAppName);
+        requireNonNull(pcjId);
+
+        // Connect to the Fluo application that is updating this instance's PCJs.
+        final AccumuloConnectionDetails cd = super.getAccumuloConnectionDetails();
+        try (final FluoClient fluoClient = new FluoClientFactory().connect(cd.getUsername(), new String(cd.getPassword()),
+                cd.getInstanceName(), cd.getZookeepers(), fluoAppName)) {
+            // Delete the PCJ from the Fluo App.
+            PeriodicQueryResultStorage periodic = new AccumuloPeriodicQueryResultStorage(getConnector(), ryaInstance);
+            DeletePeriodicQuery deletePeriodic = new DeletePeriodicQuery(fluoClient, periodic);
+            deletePeriodic.deletePeriodicQuery(pcjId, getPeriodicNotificationClient(topic, brokers));
+        }
+    }
+    
+    
+    private PeriodicNotificationClient getPeriodicNotificationClient(String topic, String brokers) throws MalformedQueryException {
+        return new KafkaNotificationRegistrationClient(topic, getProducer(brokers));
+    }
+
+    private KafkaProducer<String, CommandNotification> getProducer(String brokers) {
```

Comment:
static

Reply:
Do you guys have an opinion on if this should be called createProducer instead of getProducer since it makes a new object rather than accessing an existing?

Reply:
Yea, it should probably be renamed to createProducer.

Reply:
Renamed.  Done.